### DISCUSSION TITLE
Setup multiple routes in one code.

### DISCUSSION THREAD

**Discussion by latala43** [dotnet/yarp#846] [Q&A] [answered]
If I have a microservice with many Controllers route1, route2, ....

I have to do something like this:

```
var r1 = new ProxyRoute
{
    RouteId = "r1",
    ClusterId = "c1",
    Match = new ProxyMatch
    {
        Path = "/route1/{**catch-all}"
    }
};

var r2 = new ProxyRoute
{
    RouteId = "r2",
    ClusterId = "c1",
    Match = new ProxyMatch
    {
        Path = "/route2/{**catch-all}"
    }
};
```
The above code is working as it should be.

Meanwhile, I wish I could simplify it like the following:

```
var r = new ProxyRoute
{
    RouteId = "Routes-Of-Microservice-S1",
    ClusterId = "c1",
    Match = new ProxyMatch
    {
        Path = "/s1/{**catch-all}"
    }
};
r.WithTransformPathRemovePrefix("/s1");
```
What I understood is that 'WithTransformPathRemovePrefix("/s1");' will remove the "/s1" from the path and transforms the route to the catch all as the following:

**/s1/route1/getsomething   => /route1/getsomething
/s1/route2/getsomething   => /route2/getsomething**

Unfortunately, it returns 404 not found! Any idea how to catch everything for some prefix and route them to a specific cluster.

**Comment by Tratcher** [comment, marked as answer]
> r.WithTransformPathRemovePrefix("/s1");

Try this:
r = r.WithTransformPathRemovePrefix("/s1");

`r` is immutable so the `With*` extensions return a modified copy. This mirrors the new C#9 keyword `with`.

**Reply by latala43** [reply]
Yes. Working. Thank you
